components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -0,0 +1,624 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{
+    Checkpointer, KvEngine, OpenOptions, RaftEngine, TabletFactory, CF_DEFAULT, SPLIT_PREFIX,
+};
+use itertools::Itertools;
+use kvproto::{
+    metapb::Region,
+    raft_cmdpb::{AdminRequest, AdminResponse, RaftCmdRequest, SplitRequest},
+    raft_serverpb::RegionLocalState,
+};
+use protobuf::Message;
+use raftstore::{
+    coprocessor::split_observer::{is_valid_split_key, strip_timestamp_if_exists},
+    store::{
+        fsm::apply::extract_split_keys,
+        metrics::PEER_ADMIN_CMD_COUNTER,
+        util::{self, KeysInfoFormatter},
+        PeerStat, ProposalContext, RAFT_INIT_LOG_INDEX,
+    },
+    Result,
+};
+use slog::{info, warn, Logger};
+use tikv_util::box_err;
+
+use crate::{
+    batch::StoreContext,
+    operation::AdminCmdResult,
+    raft::{Apply, Peer},
+    router::ApplyRes,
+};
+
+#[derive(Debug)]
+pub struct SplitResult {
+    pub regions: Vec<Region>,
+    pub derived: Region,
+    pub tablet_index: u64,
+}
+
+pub fn validate_batch_split(
+    logger: &Logger,
+    req: &mut AdminRequest,
+    region: &Region,
+) -> Result<()> {
+    if !req.has_splits() || req.get_splits().get_requests().is_empty() {
+        return Err(box_err!(
+            "cmd_type is BatchSplit but it doesn't have splits request, message maybe \
+             corrupted!"
+                .to_owned()
+        ));
+    }
+    let split_reqs: &[SplitRequest] = req.mut_splits().get_requests();
+    let count = split_reqs.len();
+    for i in 1..count {
+        if split_reqs[i - 1].get_split_key() >= split_reqs[i].get_split_key() {
```

Comment:
should check split is not empty?

Reply:
`should check split_key is not empty?`
It is checked at the beginning.

Reply:
> Also should check "req.get_new_peer_ids().len() == region_to_split.get_peers().len()"

This code is to temporarily do some works of the split CoprocessorHost which does not check for it. Anyway, it will be checked in exec_batch_split.

Reply:
> > should check split_key is not empty?
> 
> It is checked at the beginning.

Do you mean this?
```
 if split_reqs.get_requests().is_empty() {
        return Err(box_err!("missing split requests"));
    }
```